### PR TITLE
test(all) increase test coverage

### PR DESCRIPTION
**Pull request by ppenalver**
## TYPE
- Tests.

## Description
- Increase test coverage
- Remove not checked validations on st-input (min, max)

### Documentation
- [x] Have explained API (Inputs, Outpus, Models, Functions, etc)
- [x] Have example running
- [x] Have example code (HTML, ts)
- [x] Have changelog.md updated

### Code
- [x] Have 3 spaces indentation
- [x] Pass ts lint Task
- [x] Pass Sass lint Task
- [x] Have QA-tags

### Tests
- [x] Have 70% Tests coverage

### CODE REVIEW

File: src/st-input/st-input.component.spec.ts [hunk 1]
```diff
@@ -22,41 +27,363 @@ describe('StInputComponent', () => {
       component.qaTag = 'test qaTag';
    });
 
-   it ('Input should has a placeholder', () => {
+   it('Input should has a placeholder', () => {
```
Comment (EvaMGC) [inline src/st-input/st-input.component.spec.ts]: Change "should has" to  "should have"

File: src/st-input/st-input.component.spec.ts [hunk 2]
```diff
@@ -22,41 +27,363 @@ describe('StInputComponent', () => {
...
+      dispatchEvent(htmlInput, 'input');
+      reactiveFixture.detectChanges();
+
+      errorMessage = reactiveFixture.debugElement.query(By.css('.error-message'));
+      expect(errorMessage).toBeDefined();
+      expect(errorMessage.nativeElement).toBeDefined();
+      expect((<HTMLSpanElement>errorMessage.nativeElement).textContent).toEqual(genericError);
+   });
+
+   it('should be notify empty error', () => {
```
Comment (EvaMGC) [inline src/st-input/st-input.component.spec.ts]: "should be notify"  to "should notify"

File: src/st-input/st-input.component.spec.ts [hunk 3]
```diff
@@ -22,41 +27,363 @@ describe('StInputComponent', () => {
...
+      dispatchEvent(htmlInput, 'input');
+      reactiveFixture.detectChanges();
+
+      errorMessage = reactiveFixture.debugElement.query(By.css('.error-message'));
+      expect(errorMessage).toBeDefined();
+      expect(errorMessage.nativeElement).toBeDefined();
+      expect((<HTMLSpanElement>errorMessage.nativeElement).textContent).toEqual('');
+   });
+
+   it('should be disable and enable', () => {
```
Comment (EvaMGC) [inline src/st-input/st-input.component.spec.ts]: Change "disable and enable" to "disabled and enabled"

File: src/st-input/st-input.component.spec.ts [hunk 4]
```diff
@@ -22,41 +27,363 @@ describe('StInputComponent', () => {
...
+      expect(htmlInput.classList).toContain('disabled');
+
+      reactiveComp.enableInput();
+      reactiveFixture.detectChanges();
+      htmlInput = reactiveFixture.debugElement.query(By.css('input')).nativeElement;
+      expect(htmlInput).toBeDefined();
+      expect(htmlInput.classList).not.toContain('disabled');
+   });
+
+   it('should be notify empty error with undefined errors object', () => {
```
Comment (EvaMGC) [inline src/st-input/st-input.component.spec.ts]: Change "should be notify" to "should notify"

File: src/st-input/st-input.component.spec.ts [hunk 5]
```diff
@@ -22,41 +27,363 @@ describe('StInputComponent', () => {
...
+      htmlInput.value = 'prueba';
+      dispatchEvent(htmlInput, 'input');
+      reactiveFixture.detectChanges();
+
+      errorMessage = reactiveFixture.debugElement.query(By.css('.error-message'));
+      expect(errorMessage).toBeNull();
+   });
+
+
+   it('should be notify no notify error with focus', () => {
```
Comment (EvaMGC) [inline src/st-input/st-input.component.spec.ts]: Change "should be notify" to "should notify"

File: src/st-search/st-search.component.spec.ts [hunk 6]
```diff
@@ -17,38 +17,43 @@ let placeholder: string = 'search a text';
...
+      comp = fixture.componentInstance;
 
-describe('StSearchComponent', () => {
+      comp.qaTag = id;
+      comp.placeholder = placeholder;
+      comp.debounce = debounce;
+      comp.minLength = minLength;
+   });
 
    it('should be init correctly', () => {
```
Comment (EvaMGC) [inline src/st-search/st-search.component.spec.ts]: Change "should be init" to "should be initialized"

File: src/st-search/st-search.component.spec.ts [hunk 7]
```diff
@@ -63,7 +68,9 @@ describe('StSearchComponent', () => {
    }));
 
    it('should be search with delay', fakeAsync(() => {
```
Comment (EvaMGC) [inline src/st-search/st-search.component.spec.ts]: Change "should be search" to "should be searched"

File: src/st-search/st-search.component.spec.ts [hunk 8]
```diff
@@ -81,27 +88,221 @@ describe('StSearchComponent', () => {
    }));
 
    it('should be search with min length', fakeAsync(() => {
```
Comment (EvaMGC) [inline src/st-search/st-search.component.spec.ts]: Change "should be search" to "should be searched"

File: src/st-search/st-search.component.spec.ts [hunk 9]
```diff
@@ -81,27 +88,221 @@ describe('StSearchComponent', () => {
...
       fixture.detectChanges();
       tick(10);
 
-      expect(outputSearch).toEqual('test');
-      tick(10);
+      expect(responseFunction).toHaveBeenCalled();
+      expect(responseFunction).toHaveBeenCalledWith('test');
    }));
+
+   it('should be search when press enter', () => {
```
Comment (EvaMGC) [inline src/st-search/st-search.component.spec.ts]: Change "should be search" to "should be searched"

File: src/st-search/st-search.component.spec.ts [hunk 10]
```diff
@@ -81,27 +88,221 @@ describe('StSearchComponent', () => {
...
+      dispatchEvent(input.nativeElement, 'input');
+      input.triggerEventHandler('keypress', { which: 13 });
+      fixture.detectChanges();
+
+      expect(responseFunction).toHaveBeenCalled();
+      expect(responseFunction).toHaveBeenCalledTimes(2);
+      expect(responseFunction).toHaveBeenCalledWith('test');
+   });
+
+   it('should be search when click', () => {
```
Comment (EvaMGC) [inline src/st-search/st-search.component.spec.ts]: Change "should be search" to "should be searched"

File: src/st-search/st-search.component.spec.ts [hunk 11]
```diff
@@ -81,27 +88,221 @@ describe('StSearchComponent', () => {
...
+      dispatchEvent(input.nativeElement, 'input');
+      searchButton.triggerEventHandler('mousedown', {});
+      fixture.detectChanges();
+
+      expect(responseFunction).toHaveBeenCalled();
+      expect(responseFunction).toHaveBeenCalledTimes(2);
+      expect(responseFunction).toHaveBeenCalledWith('test');
+   });
+
+   it('should be search only on click', () => {
```
Comment (EvaMGC) [inline src/st-search/st-search.component.spec.ts]: Change "should be search" to "should be searched"

File: src/st-search/st-search.component.spec.ts [hunk 12]
```diff
@@ -81,27 +88,221 @@ describe('StSearchComponent', () => {
...
+
+      searchButton.triggerEventHandler('mousedown', {});
+      fixture.detectChanges();
+
+      expect(responseFunction).toHaveBeenCalled();
+      expect(responseFunction).toHaveBeenCalledTimes(1);
+      expect(responseFunction).toHaveBeenCalledWith('test');
+   });
+
+   it('should be search on click twice the same', () => {
```
Comment (EvaMGC) [inline src/st-search/st-search.component.spec.ts]: Change "should be search" to "should be searched"

File: src/st-tooltip/st-tooltip.component.spec.ts [hunk 13]
```diff
@@ -65,4 +65,22 @@ describe('StTooltip', () => {
 
       expect(component.isActive).toBeFalsy();
    });
+
+   it('It can be configured to be displayed and hidden when user click and hover', () => {
```
Comment (EvaMGC) [inline src/st-tooltip/st-tooltip.component.spec.ts]: Change " when user click" to " when user clicks"

File: src/st-toggle-buttons/st-toggle-buttons.component.spec.ts [hunk 14]
```diff
@@ -1,31 +1,89 @@
...
+      expect(option3.classes['st-toggle-buttons__tab--active']).toBeFalsy();
+
+      option3.nativeElement.dispatchEvent(new Event('click'));
+      fixture.detectChanges();
+
+      expect(option1.classes['st-toggle-buttons__tab--active']).toBeFalsy();
+      expect(option3.classes['st-toggle-buttons__tab--active']).toBeTruthy();
+   });
+
+   it('should emit when change active tab on click', () => {
```
Comment (EvaMGC) [inline src/st-toggle-buttons/st-toggle-buttons.component.spec.ts]: Change test title to 'Should emit an event when active tab is clicked'

File: src/st-tab-box/st-tab-box.component.spec.ts [hunk 15]
```diff
@@ -1,31 +1,87 @@
...
+   it('should init correct', () => {
+      fixture.detectChanges();
+      let options: DebugElement[] = fixture.debugElement.queryAll(By.css('.st-tab-box__tab'));
+
+      expect(options).toBeDefined();
+      expect(options.length).toEqual(3);
+
+   });
+
+   it('should change active tab on click', () => {
```
Comment (EvaMGC) [inline src/st-tab-box/st-tab-box.component.spec.ts]: Change test title to 'Should emit an event when active tab is clicked'

File: src/st-tab-box/st-tab-box.component.spec.ts [hunk 16]
```diff
@@ -1,31 +1,87 @@
...
+      });
+
+      fixture = TestBed.createComponent(StTabBoxComponent);
+      comp = fixture.componentInstance;
+
+      comp.tabs = tabs;
+      comp.qaTag = qaTag;
+   });
+
+   it('should init correct', () => {
```
Comment (EvaMGC) [inline src/st-tab-box/st-tab-box.component.spec.ts]: Change 'correct' to correctly

File: src/st-search/st-search.component.spec.ts [hunk 17]
```diff
@@ -81,27 +88,221 @@ describe('StSearchComponent', () => {
...
+      expect(comp.focus).toBeTruthy();
+
+      dispatchEvent(input.nativeElement, 'blur');
+      fixture.detectChanges();
+
+      expect(comp.focus).toBeFalsy();
+   });
+
+
+   it('should be change initial Values', () => {
```
Comment (EvaMGC) [inline src/st-search/st-search.component.spec.ts]: Change 'should be change' to 'should be able to change'

File: src/st-search/st-search.component.spec.ts [hunk 18]
```diff
@@ -81,27 +88,221 @@ describe('StSearchComponent', () => {
...
+
+      let clearButton: DebugElement = fixture.debugElement.query(By.css('.icon-cross'));
+      expect(comp.searchBox.value).toEqual('test');
+
+      dispatchEvent(clearButton.nativeElement, 'mousedown');
+      expect(comp.searchBox.value).toEqual('');
+   });
+
+
+   it('should be change focus', () => {
```
Comment (EvaMGC) [inline src/st-search/st-search.component.spec.ts]: Change 'should be change' to 'should be able to change'

File: src/st-search/st-search.component.spec.ts [hunk 19]
```diff
@@ -81,27 +88,221 @@ describe('StSearchComponent', () => {
...
+
+      searchButton.triggerEventHandler('mousedown', {});
+      fixture.detectChanges();
+
+      expect(responseFunction).toHaveBeenCalled();
+      expect(responseFunction).toHaveBeenCalledTimes(2);
+      expect(responseFunction).toHaveBeenCalledWith('test');
+   });
+
+   it('should be clear on click', () => {
```
Comment (EvaMGC) [inline src/st-search/st-search.component.spec.ts]: Change 'should be change' to 'should be able to change'

File: src/st-input/st-input.component.spec.ts (hunk 3 above)
Comment (ppenalver) [inline src/st-input/st-input.component.spec.ts]: I change for "should be able to disable and enable"

**Pull request merged**